forcetree.c

```diff
@@ -1113,26 +1134,22 @@ void force_update_hmax(int * activeset, int size)
  *  maxnodes approximately equal to 0.7*maxpart is sufficient to store the
  *  tree for up to maxpart particles.
  */
-void force_treeallocate(int maxnodes, int maxpart)
+size_t force_treeallocate(int maxnodes, int maxpart, int first_node_offset)
```

Comment:
I was thinking of adding a new node type --  which can either point to a particle or a pseudo particle, or a internal node, or nothing.

Reply:
So these things would be actual pointers instead of these slightly ridiculous semi-pointers? I thought of that too, but the churn would have been very great at this stage, so I opted to defer it. But it sounds a good long-term goal: I found at least one place where it would allow less memory allocation and I wondered if it would let the compiler do more clever things.

Reply:
We shall still use the indices, but adding a level of indirection. See my comment below. I agree we shall defer till this is merged and correct, but good to keep this in mind while working other things out.